app/src/main/java/fr/free/nrw/commons/upload/PendingUploadsPresenter.java

```diff
@@ -0,0 +1,229 @@
+            (new PagedList.Config.Builder())
+                .setPrefetchDistance(50)
+                .setPageSize(10).build();
+        Factory<Integer, Contribution> factory;
+        boolean shouldSetBoundaryCallback;
+        if (!isSelf) {
+            //We don't want to persist contributions for other user's, therefore
+            // creating a new DataSource for them
+            contributionsRemoteDataSource.setUserName(userName);
+            factory = new Factory<Integer, Contribution>() {
+                @NonNull
+                @Override
+                public DataSource<Integer, Contribution> create() {
+                    return contributionsRemoteDataSource;
+                }
+            };
+            shouldSetBoundaryCallback = false;
+        } else {
+            contributionBoundaryCallback.setUserName(userName);
+            shouldSetBoundaryCallback = true;
+            factory = repository.fetchInProgressContributions();
+        }
+
+        LivePagedListBuilder livePagedListBuilder = new LivePagedListBuilder(factory, pagedListConfig);
+        if (shouldSetBoundaryCallback) {
+            livePagedListBuilder.setBoundaryCallback(contributionBoundaryCallback);
+        }
+
+        totalContributionList = livePagedListBuilder.build();
+    }
+
+    void getFailedContributions(String userName) {
+        final PagedList.Config pagedListConfig =
+            (new PagedList.Config.Builder())
+                .setPrefetchDistance(50)
+                .setPageSize(10).build();
+        Factory<Integer, Contribution> factory;
+        boolean shouldSetBoundaryCallback;
+        contributionBoundaryCallback.setUserName(userName);
+        shouldSetBoundaryCallback = true;
+        factory = repository.fetchFailedContributions();
+        LivePagedListBuilder livePagedListBuilder = new LivePagedListBuilder(factory,
+            pagedListConfig);
+        if (shouldSetBoundaryCallback) {
+            livePagedListBuilder.setBoundaryCallback(contributionBoundaryCallback);
+        }
+        failedContributionList = livePagedListBuilder.build();
+    }
+
+    @Override
+    public void onAttachView(@NonNull View view) {
+
+    }
+
+    @Override
+    public void onDetachView() {
+        compositeDisposable.clear();
+        contributionsRemoteDataSource.dispose();
+        contributionBoundaryCallback.dispose();
+    }
+
+    /**
+     * Delete a failed contribution from the local db
+     */
+    @Override
+    public void deleteUpload(final Contribution contribution, Context context) {
+        compositeDisposable.add(repository
+            .deleteContributionFromDB(contribution)
+            .subscribeOn(ioThreadScheduler)
+            .subscribe(() ->
+                WorkRequestHelper.Companion.makeOneTimeWorkRequest(
+                    context, ExistingWorkPolicy.KEEP)
+            ));
+    }
+
+    public void pauseUploads(List<Contribution> l, int index, Context context) {
```

Comment:
Just curious: Is there any specific reason for using one-lettered variable names? I observed this in the other PR too. In general, using more intuitive names makes the review process easier :)